ArduCopter/mode_guided.cpp

```diff
@@ -768,6 +779,163 @@ bool ModeGuided::limit_check()
     return false;
 }
 
+//loiter turns(circle) mode
+
+void ModeGuided::do_circle(const mavlink_command_long_t &packet)
+{
+    const Location circle_center = loc_from_param(packet);
+
+    // calculate radius
+    uint8_t circle_radius_m = radius_from_param(packet); // circle radius held in high byte of p1
+
+    //store the number of turns
+    turn_count = (uint8_t)packet.param1;
+
+    // move to edge of circle (verify_circle) will ensure we begin circling once we reach the edge
+    circle_movetoedge_start(circle_center, circle_radius_m);
+}
+
+Location ModeGuided::loc_from_param(const mavlink_command_long_t &packet)
+{    
+    Location ret((int32_t)packet.param5,(int32_t)packet.param6,(int32_t)packet.param7,copter.current_loc.get_alt_frame());
+
+    // use current lat, lon if zero
+    if (ret.lat == 0 && ret.lng == 0) {
+        ret.lat = copter.current_loc.lat;
+        ret.lng = copter.current_loc.lng;
+    }
+    // use current altitude if not provided
+    if (ret.alt == 0) {
+        // set to current altitude but in command's alt frame
+        int32_t curr_alt;
+        if (copter.current_loc.get_alt_cm(ret.get_alt_frame(),curr_alt)) {
+            ret.set_alt_cm(curr_alt, ret.get_alt_frame());
+        } else {
+            // default to current altitude as alt-above-home
+            ret.set_alt_cm(copter.current_loc.alt,
+                           copter.current_loc.get_alt_frame());
+        }
+    }
+    return ret;
+
+}
+
+uint8_t ModeGuided::radius_from_param(const mavlink_command_long_t &packet)
+{
+    if ((uint8_t)packet.param3 == 0) {
+        return (uint8_t)(copter.circle_nav->get_radius()/100);
```

Comment:
why this ? mavlink spec say nothing and if you never use the circle controler before it will still be 0

Reply:
@khancyr sir, the reason I added this check was that in the [discussion forum](https://discuss.ardupilot.org/t/copter-not-supporting-mav-cmd-nav-loiter-turns-question/51785)(that requested for Loiter turns in Guided Mode) the user requested an easier method to send the copter in a circle mode from the guided mode.The above check sends the copter in circle mode by just defining the number of turns.
For instance : 
- passing NAV_LOITER_TURNS 2 (use the default radius)
  
 ![Map2](https://user-images.githubusercontent.com/62841337/106563040-34f28f00-6551-11eb-9341-3436e2e67623.png)

- passing NAV_LOITER_TURNS 2 0 5 (use the radius provided,here 5m)
 ![Map3](https://user-images.githubusercontent.com/62841337/106563344-a6324200-6551-11eb-871b-aaabd6767515.png)
 (there is a difference in the radius of the circle)